ocs_ci/deployment/deployment.py

```diff
@@ -1332,18 +1333,22 @@ def deploy_ocs_via_operator(self, image=None):
                     replace_to=config.DEPLOYMENT["csv_change_to"],
                 )
 
-        # change namespace of storage system if needed
-        storage_system_data = templating.load_yaml(constants.STORAGE_SYSTEM_ODF_YAML)
-        storage_system_data["metadata"]["namespace"] = self.namespace
-        storage_system_data["spec"]["namespace"] = self.namespace
-
-        # create storage system
-        if ocs_version >= version.VERSION_4_9:
-            templating.dump_data_to_temp_yaml(
-                storage_system_data, constants.STORAGE_SYSTEM_ODF_YAML
+        if is_storage_system_needed():
+            logger.info("Creating StorageSystem")
+            # change namespace of storage system if needed
+            storage_system_data = templating.load_yaml(
+                constants.STORAGE_SYSTEM_ODF_YAML
             )
-            log_step("Apply StorageSystem CR")
-            exec_cmd(f"oc apply -f {constants.STORAGE_SYSTEM_ODF_YAML}")
+            storage_system_data["metadata"]["namespace"] = self.namespace
+            storage_system_data["spec"]["namespace"] = self.namespace
+
+            # create storage system
+            if ocs_version >= version.VERSION_4_9:
```

Comment:
I think this if block can be removed and all below moved under regular indention?

Reply:
done